### PR TITLE
Add new licenses and license updates

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
References: 
* https://github.com/aboutcode-org/scancode-toolkit/issues/3938
* https://github.com/aboutcode-org/scancode-toolkit/issues/3949
* https://github.com/aboutcode-org/scancode-toolkit/issues/3973
* https://github.com/aboutcode-org/scancode-toolkit/issues/3990
* https://github.com/aboutcode-org/scancode-toolkit/issues/3991
* https://github.com/aboutcode-org/scancode-toolkit/issues/3992
* https://github.com/aboutcode-org/scancode-toolkit/issues/3993

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁
* [x] Updated documentation pages (if applicable)
* [x] Updated CHANGELOG.rst (if applicable)

### CODE REVIEW

File: src/licensedcode/data/licenses/freetos-mit.LICENSE [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+key: freetos-mit
```
Comment (AyanSinhaMahapatra) [inline src/licensedcode/data/licenses/freetos-mit.LICENSE]: @DennisClark maybe a typo here?
```suggestion
key: freertos-mit
```

Comment (DennisClark) [reply src/licensedcode/data/licenses/freetos-mit.LICENSE]: oops

Comment (DennisClark) [reply src/licensedcode/data/licenses/freetos-mit.LICENSE]: I will fix that in DejaCode shortly

Comment (DennisClark) [reply src/licensedcode/data/licenses/freetos-mit.LICENSE]: DejaCode updates: 
* deprecated freetos-mit, setting it to inactive (by design, cannot change the license key in DejaCode)
* created freertos-mit

**Review by pombredanne — APPROVED** [review]
Looking good. Merging!

**Pull request merged**